app/src/main/java/net/gsantner/markor/activity/ActionOrderActivity.java

```diff
@@ -0,0 +1,257 @@
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+
+import net.gsantner.markor.R;
+import net.gsantner.markor.format.markdown.MarkdownTextActions;
+import net.gsantner.markor.format.plaintext.PlaintextTextActions;
+import net.gsantner.markor.format.todotxt.TodoTxtTextActions;
+import net.gsantner.markor.ui.hleditor.TextActions;
+import net.gsantner.markor.util.ActivityUtils;
+import net.gsantner.markor.util.AppSettings;
+import net.gsantner.markor.util.ContextUtils;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import static android.support.v7.widget.helper.ItemTouchHelper.ACTION_STATE_DRAG;
+
+public class ActionOrderActivity extends AppCompatActivity {
+
+    public static final String EXTRA_FORMAT_KEY = "FORMAT_KEY";
+
+    private Adapter _adapter;
+    private ArrayList<String> _keys;
+    private ArrayList<TextActions.ActionItem> _actions;
+    private TextActions _textActions;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
+            getWindow().setExitTransition(null);
+        }
+        final AppSettings appSettings = new AppSettings(this);
+        final ActivityUtils contextUtils = new ActivityUtils(this);
+        contextUtils.setAppLanguage(appSettings.getLanguage());
+        setTheme(R.style.AppTheme_Light);
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.action_order_activity);
+
+        //  Set back button
+        final Toolbar _toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(_toolbar);
+        if (getSupportActionBar() != null) {
+            getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        }
+
+        // Set up recyclerview
+        final RecyclerView recycler = findViewById(R.id.action_order_activity_recycler);
+        recycler.setLayoutManager(new LinearLayoutManager(this));
+        recycler.addItemDecoration(new DividerItemDecoration(recycler.getContext(), DividerItemDecoration.VERTICAL));
+
+        extractActionData();
+        _adapter = new Adapter(_actions);
+
+        final ItemTouchHelper.Callback callback = new ReorderCallback(_adapter);
+        final ItemTouchHelper helper = new ItemTouchHelper(callback);
+        helper.attachToRecyclerView(recycler);
+
+        recycler.setHasFixedSize(true);
+        recycler.setAdapter(_adapter);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        final MenuInflater inflater = getMenuInflater();
+        inflater.inflate(R.menu.action_order__menu, menu);
+
+        final ContextUtils cu = ContextUtils.get();
+        cu.tintMenuItems(menu, true, Color.WHITE);
+        cu.freeContextRef();
+
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Switch statements should have a default label](https://app.codacy.com/manual/gsantner/markor/pullRequest?prid=5344500)